eclair-core/src/main/scala/fr/acinq/eclair/payment/Relayer.scala

```diff
@@ -213,8 +218,8 @@ object Relayer extends Logging {
 
   // @formatter:off
   sealed trait NextPayload
-  case class FinalPayload(add: UpdateAddHtlc, payload: PerHopPayload) extends NextPayload
-  case class RelayPayload(add: UpdateAddHtlc, payload: PerHopPayload, nextPacket: OnionRoutingPacket) extends NextPayload {
+  case class FinalPayload(add: UpdateAddHtlc, payload: OnionPerHopPayload) extends NextPayload
```

Comment:
The payload isn't consistent between `FinalPayload`/`RelayPayload`. This is due to the choices made in `Onion` and I'll comment further there.